ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -108,27 +107,33 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
     : undefined;
 
   // populate default 'position' and other future properties
-  const legend =
-    props.spec.legend && validateLegendSpec(props.spec.legend)
+  const legend = useMemo(() => {
+    return props.spec.legend && validateLegendSpec(props.spec.legend)
       ? merge({}, DEFAULT_LEGEND, props.spec.legend)
       : undefined;
+  }, [props.spec.legend]);
 
   // TODO: add support for y_axis_alt.format
   const format = props.spec.yAxis?.format ?? DEFAULT_FORMAT;
 
   // ensures there are fallbacks for unset properties since most
   // users should not need to customize visual display
-  const visual = merge({}, DEFAULT_VISUAL, props.spec.visual);
+  const visual = useMemo(() => {
```

Comment:
Refactored these merge lines so they don't run unnecessarily often. Also removed useDeepMemo since it isn't actually helping and fixed up resulting eslint errors